homeassistant/components/command_line/sensor.py

```diff
@@ -152,7 +152,7 @@ def update(self) -> None:
             args = None
             args_compiled = None
         else:
-            prog, args = command.split(" ", 1)
```

Comment:
This should probably not be done this way (**security issue**)  Please use `shlex.split()` and take the first argument of the returned list for prog, the rest of the list would be args.

Good catch.

Reply:
Not a security issue -- you are supposed to be able to execute arbitrary shell commands anyway. See comments: https://github.com/akx/home-assistant-core/blob/cc41c6d10bce6c7ad0afa5f7cdba69c354918102/homeassistant/components/command_line/__init__.py#L44

Reply:
The `args` aren't passed directly on though, but via a `Template`. That could make a difference here. Either way, this PR is likely not going to go anywhere as-is due to too large a scope, so these concerns should be done in the (possible) future per-component PRs.